src/librustdoc/html/static/main.js

```diff
@@ -910,8 +908,7 @@ function defocusSearchBar() {
                 // Names didn't match so let's check if one of the generic types could.
                 if (literalSearch === true) {
                      if (obj.length > GENERICS_DATA && obj[GENERICS_DATA].length > 0) {
-                        var length = obj[GENERICS_DATA].length;
-                        for (x = 0; x < length; ++x) {
+                        for (x = 0, len = obj[GENERICS_DATA].length; x < len; ++x) {
                             if (obj[GENERICS_DATA][x] === val.name) {
                                 return true;
                             }
```

Comment:
```suggestion
if (obj[GENERICS_DATA].some(function(name) { return name === val.name; })) {
  return true;
```

Reply:
Actually, we could even use our own "onEach" function. It basically works the same after all.